### DISCUSSION TITLE
ESP32 update wont install on Zigstar UZG-01

### DISCUSSION THREAD

**Discussion by storebatfar** [xyzroe/XZG#141] [Announcements]
I have a Zigstar UZG-01 running ESP32 firmware 20240914. I am trying to install firmware 20241001. Using the UI in the device I start the update, but it just stops during download. Also tried downloading the firmware file and doing a manual update. This stops at 0%.
After several mins. I can refresh the page and the device has rebooted, but not updated the firmware.

Model | UZG-01
Firmware | 20240914
Hardware | ESP32-D0WD-V3

I normally see the update progress from 0-100%, device reboot and all ok.

**Comment by lukasHD** [comment]
same for me 👍

**Comment by Nibbes** [comment]
same here.

**Comment by Dodoooh** [comment]
Same here, but now it’s completely broken…

I tried flashing it using the web flasher:
https://xzg.xyzroe.cc/install/

However, PoE isn’t working with my ZigStar Olizig. Any ideas or suggestions?

**Comment by xyzroe** [comment]
Use 0914. And don't have any problems.

**Comment by memesalot** [comment]
@xyzroe Is there plans to keep supporting the ESP32-­SOLO-­1 (which is inside the Zigstar UZG-01)? Looking at all the latest betas they don't support it (which is probably why this issue is open).

**Comment by xyzroe** [comment]
I don't know at the moment. 
I'm working on an absolutely new version, that is based on clear idf. 
I'll try to support solo boards, but don't know how it will be implemented. 

As I know, there are just some first uzg-01 that use solo boards. About 50-100 pcs.

**Comment by memesalot** [comment]
@xyzroe I have a fairly recently purchased UZG-01 from Tindie and it has a ESP32-D0WD-V3 and cant accept the latest beta firmware. I assume this is a SOLO based one?

**Comment by xyzroe** [comment]
ESP32-D0WD-V3 is not SOLO

**Comment by victorhooi** [comment]
I have a Zigstar UZG-01, which has a `ESP32-D0WDQ6` chip.

![CleanShot 2024-12-04 at 14 13 09@2x](https://github.com/user-attachments/assets/41b6aa0f-c6d4-4003-9ec9-eca92b22e184)

I can also confirm that I was able to upgrade to the earlier `20240914` build - however, if I try to update to the latest build from October 2024, it just stalls at the `Downloading release 20241001 from Github...` step indefinitely:

![CleanShot 2024-12-03 at 22 04 14@2x](https://github.com/user-attachments/assets/013a445f-4507-4244-9784-12c2b036b6ba)

I also tried flashing with the [XZG_20241001.full.bin](https://github.com/xyzroe/XZG/releases/download/20241001/XZG_20241001.full.bin) file from the [XZG Github Releases page](https://github.com/xyzroe/XZG/releases), and that also seems to just hang as well.

**Comment by xyzroe** [comment]
If you want to install BETA version you need to use *.ota file form releases. Not full one.

**Comment by jacobhallgren** [comment]
![image](https://github.com/user-attachments/assets/a456158b-9c52-46e9-aacb-3960851e9340)
I cannot upgrade my device from 2024-05-19 to 2024-09-14. It just keeps loading.

**Reply by SyBeer** [reply]
![image](https://github.com/user-attachments/assets/9c40c4c6-32be-43ee-84d0-bcaf699321dd)
It works in my case, but it won't flash the first time. 
I’m using Safari on Mac OS X, and the flashing was done via Etherne

**Comment by pullasuti** [comment]
> If you want to install BETA version you need to use *.ota file form releases. Not full one.

So that's why the flashing fails? The built-in webui flasher is also trying to flash the full .bin file for the beta.

Although flashing the ota file seems to have bricked my D0WD for now.

**Comment by victorhooi** [comment]
OK, so I was able to load the `[XZG_20241001.ota.bin](https://github.com/xyzroe/XZG/releases/download/20241001/XZG_20241001.ota.bin)` file, as suggested by @xyzroe.

However, now when I try to load the web UI for the UZG-01, I get a HTTP 404 with the text:

`File not found`

![CleanShot 2024-12-13 at 19 31 47@2x](https://github.com/user-attachments/assets/bb26e11f-abdc-40ac-a59d-03d7072eb4b7)

Is this a bug in the latest beta release? Or did I do something wrong?

**Comment by xyzroe** [comment]
It's some kind of bug. 
Web UI files must be downloaded automatically while the first boot of XZG, but sometimes it doesn't work. 
So you have two options:
1. Reflash using a USB cable and web flasher. 
2. Upload *.fs.bin manually using PostMan. You can find details on the screenshot. 


![SAVE_20241213_093751.jpg](https://github.com/user-attachments/assets/6d2b5f0f-490d-4e6d-9411-7cc176c94e38)

**Comment by yyax** [comment]
![image](https://github.com/user-attachments/assets/d99f9b84-8068-44db-b59c-25728772c67f)

**Comment by tangelogh** [comment]
Same here, stuck on firmware version 20240707..
Wont try to force flash an ota file when it can fail so easily..
Any ETA or even idea on whats the issue?
I can test some things if you want, but kinda sad that I paid almost 50€ for a device with such a conturbated life cycle (started as a UZG, then XZG, then stuck in a random old firmware)..

**Comment by rickzed** [comment]
what can i do for this ![屏幕截图 2025-01-05 200656](https://github.com/user-attachments/assets/8d6b5ac9-85f4-4bcf-b361-4b4cf562c3c5)

**Comment by mihsu81** [comment]
Why don't you flash it over USB using the web flasher? Had the same issue a few months ago and in 10 minutes I got it flashed with the latest beta and reconfigured.

**Comment by Bixkitts** [comment]
`pio run -t upload`
Also results in no web files being uploaded to flash over USB for me (fresh clone of the repo).

**Comment by xyzroe** [comment]
Just use version 0914. And don't have any problems. 

Or install 1001 using a web flasher.

**Comment by xyzroe** [comment]
> what can i do for this ![屏幕截图 2025-01-05 200656](https://github.com/user-attachments/assets/8d6b5ac9-85f4-4bcf-b361-4b4cf562c3c5)
> 


looks like some error in URL. 
Or you're trying on another version than 1001.

**Comment by xyzroe** [comment]
> Same here, stuck on firmware version 20240707..
> Wont try to force flash an ota file when it can fail so easily..
> Any ETA or even idea on whats the issue?
> I can test some things if you want, but kinda sad that I paid almost 50€ for a device with such a conturbated life cycle (started as a UZG, then XZG, then stuck in a random old firmware)..

In your situation you need to use 0914*.ota.bin file and update manually through the web UI.

**Comment by xyzroe** [comment]
> Any ETA or even idea on whats the issue?
> I can test some things if you want, but kinda sad that I paid almost 50€ for a device with such a conturbated life cycle (started as a UZG, then XZG, then stuck in a random old firmware)..

Are you seriously complaining about the price you paid for hardware while using free software that is provided with absolutely no guarantees, on an as-is basis? No one forced you to buy the device if you weren't prepared to accept these terms. Instead of whining, spend your time at least trying to figure things out and testing some solutions yourself. After all, hundreds of users are satisfied and have no issues.

**Comment by ADMiNZ** [comment]
<img width="1255" alt="2025-01-06_00-36-36" src="https://github.com/user-attachments/assets/cdf3643b-bba0-433c-aaa6-7aa7843f3210" />

The file is a problem. But that does not help. I also have a favorable connection with the device.

**Comment by xyzroe** [comment]
> <img width="1255" alt="2025-01-06_00-36-36" src="https://github.com/user-attachments/assets/cdf3643b-bba0-433c-aaa6-7aa7843f3210" />
> 
> The file is a problem. But that does not help. I also have a favorable connection with the device.


You need to use the *.fs.bin file if there isn't a web UI after the update.

**Comment by ADMiNZ** [comment]
> You need to use the *.fs.bin file if there isn't a web UI after the update.


Hello. I have passed by the web interface - the system and overloads international communication or international communication (with ping there are losses in the same rhythm.)

**Comment by xyzroe** [comment]
Looks like incorrect hardware identification. Use version 0914

**Comment by danielszilagyi** [comment]
I tried to send the filesystem image with POST to both an UZG-01 and Tube's ZB PoE (read this thread after PR submitted) but getting instant reboot. On the console there is a message about protected flash (I will try to get it exactly later) then guru meditation and reboot.
On the Tube's ZB I tried different versions, when flashing my build I get the 404 error, when flashing 20240914 I have the UI.
Also on Tube's ZB the filesystem is shown as 60 kB large.

I now understand why the UI doesn't have the new localization if it is downloaded at first boot. But on the UZG-01 it looks like the filesystem remained containing the old files (without my localization)

Tried to flash the full image on Tube's ZB with ESPHome-flasher but getting error: `Unexpected error: The firmware binary is invalid (magic byte=FF, should be E9)` with all the full images, not just the one I built.

What is it that I am missing?

**Reply by xyzroe** [reply]
Different partitions layout. Use USB mode to flash 1001. But keep in mind that it is still beta.

**Reply by danielszilagyi** [reply]
I flashed the full bin tonight to the device via USB, but I am still missing the [translation](https://github.com/xyzroe/XZG/pull/154) I added.

```shell
$ strings XZG_20250302.full.bin |grep hu.jso
/json/hu.json.gz
/lg/hu.json
hu.json.gz
```

This is from the file I flashed.
When I try to GET /lg/hu.json I get the 404.

Any idea what could be missing?

**Reply by xyzroe** [reply]
https://github.com/xyzroe/XZG/blob/main/tools%2Fwebfilesbuilder%2Fbuild_html.py

You must run this script to build a file system

**Reply by danielszilagyi** [reply]
I built using the vscode tooling, and the resulting binaries contain the localized files. I also saw that the tooling does run build_html.py as a pre-req step. The compressed json is there, everything is there when I check the build.
I flashed the full .bin via USB after building.
I can GET all the localized files (those are on the filesystem, right?) except the one I newly added.
UI works as expected. Just the PR modifications seems to be missing.
I could not re-mount the filesystem image to verify from another machine though.

Do I need to send the fs image via POST after I flashed the full binary?

**Reply by xyzroe** [reply]
No

**Comment by strelok** [comment]
I have ZigStar LilyZig (POE).. flashed latest 20241001 and got 'file not found' in web.
Then uploaded XZG_20241001.fs.bin manually, restarted device but still same issue. 
Now trying to use web flasher via usb but it's not recognizing as a device at all. Tried windows and mac os, tried several different cables, drivers are installed etc 
What I'm missing? please help :)

**Reply by strelok** [reply]
uploaded XZG_20241001.fs.bin once again and still nothing :)

**Reply by xyzroe** [reply]
That's cause of different fs layout. The only way is to flash using USB. I have just checked web flasher - it works. So you should check you hardware.

**Reply by strelok** [reply]
@xyzroe 
ZigStar LilyZig worked just fine a few years... do you know something about this device? any ideas why it's not recognized and can't be flashed.. ?

**Comment by dev-goalz** [comment]
WebFlash didn't work, POSTman give the "Upload OK" but no results.

This POST worked for me, look at the older version bin!

`curl -X POST -F "file=@XZG_20240914.ota.bin" http://<IP>/update`

Did run this command for over 10x before it worked, even successful flashes do not always work.
I think this is a race condition with the bootloop.

**Reply by et1783** [reply]
>

>  I think this is a race condition with the bootloop.

That's the key point!!!!!!!!!!!!!!!!!!!! Finally Succussed after continuing POST for ten more times that got 200 response with 0914 OTA!

**Comment by victorhooi** [comment]
Hmm, I'm still hitting some issues trying to reflash my UZG-01, to get the web UI working.

I did try pushing an update over HTTP POST to /update, using `XZG_20241001.fs.bin` - however, after a minute or so, the HTTP request failed:

![CleanShot 2025-07-01 at 18 13 30@2x](https://github.com/user-attachments/assets/eb366b7c-a126-4334-9f7f-911a3741a0f0)

I then tried connecting the UZG-01 to my computer via USB cable, and flashing using the XYZ web flasher (https://xzg.xyzroe.cc/install/) and version `20241001`:

However, I get an error message about `Failed to initialize. Try resetting your device or holding the BOOT button while clicking INSTALL.':

![CleanShot 2025-07-01 at 18 28 05@2x](https://github.com/user-attachments/assets/8f6cf6bc-1c9a-4ae8-8dd6-df5b865ff127)

I did try resetting my device, by holding down the control button for 4-10 seconds (per the [button instructions here](https://uzg.zig-star.com/configuration/#configuring-with-button)) - however, when I then try to flash again, I get the same error message about `Failed to initialize`.

@xyzroe Did I perhaps miss a step here somewhere in the flashing? Or do you have any advice on how to re-flash my device please?

**Reply by xyzroe** [reply]
check COM port twice, check drivers, check cable.

**Comment by victorhooi** [comment]
Hmm, I think I got it working - based on the comment [here](https://github.com/xyzroe/XZG/discussions/141#discussioncomment-12926848) - I don't know enough to confirm if it's a race condition as they say, but retrying a few times seems to work?

Anyway, I was able to finally revert back to the earlier firmware using this:
```
curl -X POST -F "file=@XZG_20240914.ota.bin" http://10.5.1.10/update
```
Then, via the XZG web UI, I can confirm I was able to update using `XZG_20241001.ota.bin`.

The web UI was still broken after that, so I fixed it using XYZ's suggestion - had to retry it:
```
curl -X POST -F "file=@XZG_20241001.fs.bin" http://10.5.1.10/update
```
There's definitely a bit of mucking around - but I guess maybe it's because the XYZ firmware is in a bit of flux or something at the moment?

@xyzroe You mentioned in December last year that you were doing some work to update the IDF version - is that right? =) Just curious how that was going, or what the current state of things in XYZ firmware is? Definitely excited to see what's coming up!

**Reply by xyzroe** [reply]
You just shouldn't use 1001 if you don't want to look for solutions to some of the complexities. the best way to install 1001 is via USB, as it changes the partition table a bit.

**Reply by xyzroe** [reply]
No work on a new version of the firmware is not underway, unfortunately.
I think the necessary functionality is performed by those versions that are available at the moment.

I started to work on a new version, on pure IDF, but unfortunately due to changed personal circumstances, and the absence of any noticeable support from users (or device manufacturers), I do not have time to do it.

**Reply by rodionos** [reply]
It seems I have a similar issue. I Installed 20241001 version using the web flasher (serial) on my hamgeek POE coordinator.

```
8878 [printLogMsg] [RCP] FW: 20240710
8890 [setup] [RCP] FW: 20240710
8890 [setup] [ESP] FW: 20241001
8890 [setupCoordinatorMode] Network
9009 [mDNS_start] mDNS responder started on XZG.local
9044 [setup] 
{
  "system-vars": {
    "hwBtnIs": true,
    "hwLedUsbIs": true,
    "hwLedPwrIs": true,
    "hwZigbeeIs": true,
    "connectedClients": 0,
    "socketTime": 0,
    "connectedEther": false,
    "apStarted": false,
    "wifiWebSetupInProgress": false,
    "vpnWgInit": false,
    "vpnWgConnect": false,
    "vpnWgPeerIp": "0.0.0.0",
    "vpnWgCheckTimeKey": 0,
    "vpnHnInit": false,
    "mqttConn": false,
    "mqttReconnectTime": 0,
    "mqttHeartbeatTime": 0,
    "disableLeds": false,
    "zbFlashing": false,
    "deviceId": "XZG-D123",
    "espUpdAvail": false,
    "rcpUpdAvail": false
  }
```

After powering with POE, the device is visible on the local network. The UI fails to show up, and an attempt to upload web files with curl fails either with a timeout or reset.

```
curl -X POST -F "file=@XZG_20241001.fs.bin" http://x.x.x.x/update
curl: (56) Recv failure: Connection reset by peer
```

@xyzroe - What is the recommended way of downgrading an earlier version which does not have these UI issues? I may be located in a network where the coordinator may not be able to download arbitrary files.

**Reply by xyzroe** [reply]
Using POST on http://x.x.x.x/update or USB based web flasher on the web site.

**Reply by rodionos** [reply]
Web flashing to 20240914 fixed the problem. If I had to use the `/update`endpoint, should it be full or ota file for the given release?

**Reply by xyzroe** [reply]
for 20241001 you should use separate file .fs.bin

**Comment by jeblove** [comment]
I upgraded the device to version 20241001 and encountered the same "file not found" issue.

I tried using a POST request to upload "fs.bin" and re-flash the firmware, but received an "Error" message. Only uploading "ota.bin" resulted in "Upload OK".  
At this point, the device's yellow indicator light started blinking, and the web interface became accessible.  
However, after about one or two minutes, the yellow light turned into a steady glow, and accessing the web interface again resulted in "file not found".

My final solution was to first flash "20240914.ota.bin", and then immediately re-flash "XZG_20240914.full.bin".
